### ISSUE TITLE
Failed to push changes!

### ISSUE THREAD

**Issue by Nageswari-droid** [neel1996/gitconvex#101]
```

2020/11/17 15:57:26 INFO: Pushing changes to remote -> github : refs/heads/master:refs/heads/master
runtime error: invalid memory address or nil pointer dereference

goroutine 304 [running]:
runtime/debug.Stack(0x1, 0x0, 0x0)
        c:/go/src/runtime/debug/stack.go:24 +0xa5
runtime/debug.PrintStack()
        c:/go/src/runtime/debug/stack.go:16 +0x29
github.com/99designs/gqlgen/graphql.DefaultRecover(0xedcc40, 0xc0001096e0, 0xd20da0, 0x12a64a0, 0x0, 0x0)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/recovery.go:16 +0xae
github.com/99designs/gqlgen/graphql.(*OperationContext).Recover(0xc0005f0800, 0xedcc40, 0xc0001096e0, 0xd20da0, 0x12a64a0, 0xaed90d, 0x1cd29a416e0)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/context_operation.go:110 +0x5d
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation_pushToRemote.func1(0xc00067ee00, 0xc00067ee80, 0xc0000a14e8)
        C:/Users/SVNPC/Desktop/js/gitconvex/graph/generated/generated.go:3182 +0x85
panic(0xd20da0, 0x12a64a0)
        c:/go/src/runtime/panic.go:969 +0x1c7
github.com/go-git/go-git/v5/plumbing/transport/ssh.(*PublicKeysCallback).ClientConfig(0x0, 0xc000174510, 0x0, 0x1cd29a40f70)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/transport/ssh/auth_method.go:232 +0x2d
github.com/go-git/go-git/v5/plumbing/transport/ssh.(*command).connect(0xc0005ac320, 0xed3780, 0x0)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/transport/ssh/common.go:113 +0x58
github.com/go-git/go-git/v5/plumbing/transport/ssh.(*runner).Command(0xc000006618, 0xdf4886, 0x10, 0xc0003a4780, 0xed3780, 0x0, 0x2, 0x3, 0x0, 0xc0000b57a0)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/transport/ssh/common.go:52 +0x96
github.com/go-git/go-git/v5/plumbing/transport/internal/common.(*client).newSession(0xc0000f85e0, 0xdf4886, 0x10, 0xc0003a4780, 0xed3780, 0x0, 0x0, 0x0, 0xc0003a4780)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/transport/internal/common/common.go:112 +0x90
github.com/go-git/go-git/v5/plumbing/transport/internal/common.(*client).NewReceivePackSession(0xc0000f85e0, 0xc0003a4780, 0xed3780, 0x0, 0xc0003a4780, 0x0, 0x0, 0xc0005bc260)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/transport/internal/common/common.go:96 +0x69
github.com/go-git/go-git/v5.newSendPackSession(0xc00036c900, 0x2e, 0xed3780, 0x0, 0xd27940, 0x5d0920, 0xc0000a0f58, 0xc0005a2c18)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/remote.go:387 +0xaa
github.com/go-git/go-git/v5.(*Remote).PushContext(0xc000315120, 0xedcbc0, 0xc00002c0c0, 0xc0000a1208, 0x0, 0x0)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/remote.go:105 +0x1fb
github.com/go-git/go-git/v5.(*Remote).Push(...)
        C:/Users/SVNPC/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/remote.go:87
github.com/neel1996/gitconvex-server/git.PushToRemote(0xc000174090, 0xc00025e480, 0x6, 0xc00039008a, 0x6, 0x58417d, 0xdeb24f)
        C:/Users/SVNPC/Desktop/js/gitconvex/git/git_push.go:26 +0x366
github.com/neel1996/gitconvex-server/graph.(*mutationResolver).PushToRemote(0xc000326100, 0xedcc40, 0xc0001096e0, 0xc000390038, 0x8, 0xc000390050, 0x2e, 0xc00039008a, 0x6, 0xc000109740, ...)
        C:/Users/SVNPC/Desktop/js/gitconvex/graph/schema.resolvers.go:113 +0x14a
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation_pushToRemote.func2(0xedcc40, 0xc0001096e0, 0x20, 0xd81700, 0xc00067ef01, 0xc0003209c0)
        C:/Users/SVNPC/Desktop/js/gitconvex/graph/generated/generated.go:3204 +0x222
github.com/99designs/gqlgen/graphql/executor.processExtensions.func3(0xedcc40, 0xc0001096e0, 0xc0003209c0, 0xc000109710, 0xc000109740, 0x0, 0x0)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/extensions.go:62 +0x42
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation_pushToRemote(0xc00067ee00, 0xedcc40, 0xc0001096b0, 0xc000664300, 0x0, 0x0, 0x0, 0x0, 0x0)
        C:/Users/SVNPC/Desktop/js/gitconvex/graph/generated/generated.go:3202 +0x2d2
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Mutation(0xc00067ee00, 0xedcc40, 0xc0001096b0, 0xc0005bc0d0, 0x1, 0x1, 0xc000109680, 0x30)
        C:/Users/SVNPC/Desktop/js/gitconvex/graph/generated/generated.go:6214 +0x9cf
github.com/neel1996/gitconvex-server/graph/generated.(*executableSchema).Exec.func2(0xedcc40, 0xc000109680, 0x10)
        C:/Users/SVNPC/Desktop/js/gitconvex/graph/generated/generated.go:935 +0x85
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1.1(0xedcc40, 0xc000109680, 0xc00067ee40)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/executor.go:105 +0x4a
github.com/99designs/gqlgen/graphql/executor.processExtensions.func2(0xedcc40, 0xc000109680, 0xc00067ee40, 0xebba90)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/extensions.go:59 +0x42
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1(0xedcc40, 0xc000109620, 0xc0001095f0)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/executor.go:104 +0x131
github.com/99designs/gqlgen/graphql/handler/transport.POST.Do(0xedad80, 0xc0000ee000, 0xc0005f0700, 0xed92c0, 0xc0000debd0)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/handler/transport/http_post.go:53 +0x3c8
github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP(0xc000225c00, 0xedad80, 0xc0000ee000, 0xc0005f0700)
        C:/Users/SVNPC/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/handler/server.go:115 +0x21b
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0002c4600, 0xedad80, 0xc0000ee000, 0xc0005f0000)
        C:/Users/SVNPC/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xe5
github.com/rs/cors.(*Cors).Handler.func1(0xedad80, 0xc0000ee000, 0xc0005f0000)
        C:/Users/SVNPC/go/pkg/mod/github.com/rs/cors@v1.7.0/cors.go:219 +0x1c6
net/http.HandlerFunc.ServeHTTP(0xc000225d80, 0xedad80, 0xc0000ee000, 0xc0005f0000)
        c:/go/src/net/http/server.go:2042 +0x4b
net/http.serverHandler.ServeHTTP(0xc0000ee1c0, 0xedad80, 0xc0000ee000, 0xc0005f0000)
        c:/go/src/net/http/server.go:2843 +0xaa
net/http.(*conn).serve(0xc000086000, 0xedcb80, 0xc0002c2740)
        c:/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
        c:/go/src/net/http/server.go:2969 +0x36d
```

**Comment by neel1996** [comment]
@Nageswari-droid fix pushed as part of 8ff019663967452cde79db80399423545683c5e9

**Comment by Nageswari-droid** [comment]
Working!

**Comment by Nageswari-droid** [comment]
push throws error while displaying un-pushed commits for a new repo 

```
runtime error: invalid memory address or nil pointer dereference

goroutine 7 [running]:
runtime/debug.Stack(0x1, 0x0, 0x0)
        c:/go/src/runtime/debug/stack.go:24 +0xa5
runtime/debug.PrintStack()
        c:/go/src/runtime/debug/stack.go:16 +0x29
github.com/99designs/gqlgen/graphql.DefaultRecover(0xfa1ba0, 0xc000598930, 0xde4ea0, 0x136c4a0, 0x0, 0x0)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/recovery.go:16 +0xae
github.com/99designs/gqlgen/graphql.(*OperationContext).Recover(0xc000092e00, 0xfa1ba0, 0xc000598930, 0xde4ea0, 0x136c4a0, 0x0, 0xcc93ef)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/context_operation.go:110 +0x5d
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Query_gitUnPushedCommits.func1(0xc00020e710, 0xc00020e760, 0xc0005b1368)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/generated/generated.go:3775 +0x85
panic(0xde4ea0, 0x136c4a0)
        c:/go/src/runtime/panic.go:969 +0x1c7
github.com/neel1996/gitconvex-server/git.UnPushedCommits.func1(0xc0002b2690, 0xc0002b2690, 0x0)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/git/git_unpushed_commits.go:74 +0x4f
github.com/go-git/go-git/v5/plumbing/object.(*commitPostIterator).ForEach(0xc0001beee0, 0xc0001bef00, 0xf9e4a0, 0xc0001beee0)
        C:/Users/neelu/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/object/commit_walker.go:173 +0x69
github.com/neel1996/gitconvex-server/git.UnPushedCommits(0xc000598ab0, 0xc0005b10e8, 0xd, 0xeac3c6, 0x1, 0xc0001d0b1f)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/git/git_unpushed_commits.go:73 +0x2fc
github.com/neel1996/gitconvex-server/graph.(*queryResolver).GitUnPushedCommits(0xc000686280, 0xfa1ba0, 0xc000598930, 0xc0001d0ac6, 0x8, 0xc0001d0add, 0x30, 0xc0001d0b1f, 0x6, 0xde0840, ...)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/schema.resolvers.go:296 +0x22d
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Query_gitUnPushedCommits.func2(0xfa1ba0, 0xc000598930, 0x20, 0xe45960, 0xc00020e701, 0xc0001be160)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/generated/generated.go:3797 +0x228
github.com/99designs/gqlgen/graphql/executor.processExtensions.func3(0xfa1ba0, 0xc000598930, 0xc0001be160, 0xc000598960, 0xc000598990, 0x0, 0x0)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/extensions.go:62 +0x42
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Query_gitUnPushedCommits(0xc00020e710, 0xfa1ba0, 0xc000598900, 0xc0000ce880, 0x0, 0x0, 0x0, 0x0, 0x0)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/generated/generated.go:3795 +0x2d9
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Query.func10(0x0, 0x0)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/generated/generated.go:6416 +0xaf
github.com/99designs/gqlgen/graphql.(*FieldSet).Dispatch(0xc00050c960)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/fieldset.go:34 +0x1ed
github.com/neel1996/gitconvex-server/graph/generated.(*executionContext)._Query(0xc00020e710, 0xfa1ba0, 0xc0005988d0, 0xc00020e3f0, 0x1, 0x1, 0xc0005988d0, 0x30)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/generated/generated.go:6486 +0x452
github.com/neel1996/gitconvex-server/graph/generated.(*executableSchema).Exec.func1(0xfa1ba0, 0xc0005988d0, 0x10)
        D:/bkup/GitConvex/gitconvex_v2.0.0/gitconvex/graph/generated/generated.go:921 +0x85
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1.1(0xfa1ba0, 0xc0005988d0, 0xc00020e720)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/executor.go:105 +0x4a
github.com/99designs/gqlgen/graphql/executor.processExtensions.func2(0xfa1ba0, 0xc0005988d0, 0xc00020e720, 0xf808d0)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/extensions.go:59 +0x42
github.com/99designs/gqlgen/graphql/executor.(*Executor).DispatchOperation.func1.1(0xfa1ba0, 0xc000598870, 0xc000598840)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/executor/executor.go:104 +0x131
github.com/99designs/gqlgen/graphql/handler/transport.POST.Do(0xf9fce0, 0xc0004cc1c0, 0xc000092d00, 0xf9e220, 0xc0000aeab0)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/handler/transport/http_post.go:53 +0x3c8
github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP(0xc0002255a0, 0xf9fce0, 0xc0004cc1c0, 0xc000092d00)
        C:/Users/neelu/go/pkg/mod/github.com/99designs/gqlgen@v0.13.0/graphql/handler/server.go:115 +0x21b
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0002c6600, 0xf9fce0, 0xc0004cc1c0, 0xc000092a00)
        C:/Users/neelu/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xe5
github.com/rs/cors.(*Cors).Handler.func1(0xf9fce0, 0xc0004cc1c0, 0xc000092a00)
        C:/Users/neelu/go/pkg/mod/github.com/rs/cors@v1.7.0/cors.go:219 +0x1c6
net/http.HandlerFunc.ServeHTTP(0xc0002257e0, 0xf9fce0, 0xc0004cc1c0, 0xc000092a00)
        c:/go/src/net/http/server.go:2042 +0x4b
net/http.serverHandler.ServeHTTP(0xc0000c01c0, 0xf9fce0, 0xc0004cc1c0, 0xc000092a00)
        c:/go/src/net/http/server.go:2843 +0xaa
net/http.(*conn).serve(0xc0001030e0, 0xfa1ae0, 0xc0002cb880)
        c:/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
        c:/go/src/net/http/server.go:2969 +0x36d
```

**Comment by neel1996** [comment]
@Nageswari-droid fix pushed as part of be7a403

**Comment by neel1996** [comment]
additional fix applied -> https://github.com/neel1996/gitconvex/commit/5d6cc130e9ee1f25e1bc5aff0e74d14b5fd89a1b
